### ISSUE TITLE
Projectile breaks when fish is the default shell.

### ISSUE THREAD

**Issue by cjolowicz** [syl20bnr/spacemacs#12221]
#### Description :octocat:
Projectile breaks when `fish` is the default shell.

People have reported this problem on gitter and reddit:

- https://gitter.im/syl20bnr/spacemacs?at=58ce590f4ec6efa73051b29b
- https://www.reddit.com/r/spacemacs/comments/ag32lf/could_you_help_me_fix_projectile_it_stopped/

#### Reproduction guide :beetle:
- Make `fish` the default shell.
- Start Emacs.
- Hit `SPC p p RET`.

*Observed behaviour:* :eyes: :broken_heart:
The following error message is displayed, instead of switching to the project:

```
In ‘Projectile files’ source: ‘#[0 \301 \205;r\302!\203\206p\211q\210\303 \304 \305\211:\2036@\262\306"BB\262A\262\202\211\237\266\204)\207 [helm-current-buffer projectile-project-p buffer-live-p projectile-project-root projectile-current-project-files nil expand-file-name] 8]’ 
 (file-error "Setting current directory" "File name too long" "/Users/cjolowicz/.dotfiles/bind: Key with name 'right' does not have any mapping
bind: Key with name 'left' does not have any mapping
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'btab' does not have any mapping
bind: Key with name 'down' does not have any mapping
bind: Key with name 'up' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'right' does not have any mapping
bind: Key with name 'left' does not have any mapping
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'btab' does not have any mapping
bind: Key with name 'down' does not have any mapping
bind: Key with name 'up' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'right' does not have any mapping
bind: Key with name 'left' does not have any mapping
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'btab' does not have any mapping
bind: Key with name 'down' does not have any mapping
bind: Key with name 'up' does not have any mapping
bind: Key with name 'f1' does not have any mapping
/")
```

*Expected behaviour:* :heart: :smile:
Spacemacs switches to the project under point.

#### System Info :computer:
- OS: darwin
- Emacs: 26.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 67e969072)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(nginx docker python
       (javascript :variables js2-basic-offset 2 js-indent-level 2)
       yaml shell-scripts helm auto-completion better-defaults emacs-lisp git markdown multiple-cursors treemacs org osx react
       (shell :variables shell-default-height 30 shell-default-position 'bottom)
       spell-checking syntax-checking version-control xclipboard)
```
- System configuration features: JPEG RSVG IMAGEMAGICK GLIB NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS LCMS2

#### Backtrace :paw_prints:
```
N/A
```

**Comment by eccentric-j** [comment]
Experiencing this issue as well found some insight from others that have experienced it: https://www.reddit.com/r/spacemacs/comments/ag32lf/could_you_help_me_fix_projectile_it_stopped/

Trying that solution now.

**Comment by duianto** [comment]
The source might:
>you have an old version of the fish_vi_key_bindings function

source:
vi mode prints errors in Emacs when TERM=dumb https://github.com/fish-shell/fish-shell/issues/5806

**Comment by eccentric-j** [comment]
```
fish --version                                                 
fish, version 3.0.2
```

Seems I'm running the latest fish release so I take it that fix hasn't been merged?

**Comment by duianto** [comment]
I don't know, I just tried looking for this issue on the fish-shell repository, and saw that the same error messages are shown both here and there.

That issue has a link to this issue:
Strange argv handling in fish_vi_key_bindings https://github.com/fish-shell/fish-shell/issues/5028

Which is closed and marked with milestone fish-3.0.

So I would suspect that it's part of 3.0.

**Comment by cjolowicz** [comment]
This workaround fixes the issue for me: `fish_vi_key_bindings 2>/dev/null` (in the fish startup file).

So the question is, how does the stderr output of the shell startup end up being interpreted as a directory?

It looks like emacs is calling out to the shell, maybe to determine some environment variables as set by shell startup scripts, and ends up effectively redirecting stderr to stdout?
